app/controller/command/commands/export.py

```diff
@@ -0,0 +1,192 @@
+            present_subcommands = [subcommand for subcommand in
+                                   all_subcommands
+                                   if subcommand in command_arg]
+            present_subcommand = None
+            if len(present_subcommands) == 1:
+                present_subcommand = present_subcommands[0]
+            return self.get_help(subcommand=present_subcommand), 200
+
+        if args.which == "emails":
+            try:
+                user_command = self.facade.retrieve(User, user_id)
+
+                # Check if team name is provided
+                if args.team is not None:
+                    if user_command.permissions_level == \
+                            Permissions.team_lead or Permissions.admin:
+                        teams = self.facade.query_or(
+                            Team, [('github_team_name', str(args.team))])
+
+                        if len(teams) > 1:
+                            return self.multiple_team_same_name_msg, 200
+
+                        if len(teams) == 0:
+                            return self.no_team_found_msg, 200
+
+                        params = [('github_user_id', ids)
+                                  for ids in list(teams[0].members)]
+                        users = self.facade.query_or(User, params)
+                        return self.export_emails_helper(users)
+                    else:
+                        return self.permission_error, 200
+                else:  # if team name is not provided, export all emails
+                    if user_command.permissions_level == Permissions.admin:
+                        users = self.facade.query(User)
+                        return self.export_emails_helper(users)
+                    else:
+                        return self.permission_error, 200
+            except LookupError:
+                return self.lookup_error, 200
+        else:
+            return self.get_help(), 200
+
+    def export_emails_helper(self,
+                             users: list) -> ResponseTuple:
+        """
+        1. if team name not provided -> export emails of all users
+         as a string + names of the users who do not have an email
+
+        2. if team name provided -> export emails of all members of
+         the team + names of the users who do not have an email
+
+        """
+
+        if len(users) == 0:
+            return self.no_user_msg, 200
+
+        emails = []
+        ids_missing_emails = []
+
+        for i in range(len(users)):
+            if users[i].email == '':
+                ids_missing_emails.append(users[i].slack_id)
+                continue
+
+            emails.append(users[i].email)
+
+        # Check char count. Slack currently allows to send 16000 characters max
+        emails_str = ",".join(emails)
+
+        # Approximate char count for the names (20 chars)
+        # of the people who are missing emails
+        missing_emails_char_count = len(ids_missing_emails) * 20
```

Comment:
Please don't approximate. I/O costs are orders of magnitude more than computation costs. See comment below.